qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java

```diff
@@ -20,95 +24,132 @@
+		prefix = RandomStringUtils.randomAlphanumeric(6);
+		testPatient = createTestPatient();
+		new TestData.TestVisit(testPatient.uuid, VISIT_TYPE_UUID, getLocationUuid(homePage)).create();
 	}
 
 	@After(RunTest.HOOK.SELENIUM_PATIENT_DEMOGRAPHICS)
 	public void destroy() {
 		deletePatient(testPatient);
 		quit();
 	}
+	
+	@Given("a user clicks on the find patient app on the home page")
+	public void clickOnFindPatientRecordApp() {
+		findPatientPage = homePage.goToFindPatientRecord();
+	}
+	
+	@Given("a user clicks on the active visits app on the home page")
+	public void clickOnActiveVisitsApp() {
+		activeVisitsPage = homePage.goToActiveVisitsSearch();
+	}
+	
+	@Then("the system loads the active visits page")
+	public void systemLoadsActiveVisitsPage() {
+		assertTrue(textExists("Active Visits"));
+	}
+	
+	@Then("the system loads the find patient page")
+	public void systemLoadsFindPatientPage() {
+		assertTrue(textExists("Find Patient Record"));
+	}
+	
+	@And("a user searches for the patient to edit")
+	public void searchForPatientToEdit() {
+		findPatientPage.enterPatient(testPatient.identifier);
+	}
+	
+	@And("a user searches for the active visit patient to edit")
+	public void searchForActiveVisitPatientToEdit() {
+		activeVisitsPage.search(testPatient.identifier);
+	}
+	
+	@And("a user clicks on the first patient")
+	public void clickOnTheFirstPatient() {
+		dashboardPage = findPatientPage.clickOnFirstPatient();
+	}
+	
+	@And("a user clicks on the active visit patient to edit")
+	public void clickOnActivePatientToEdit() {
+		dashboardPage = activeVisitsPage.goToPatientDashboardOfLastActiveVisit();
+	}
 
-	@Given("a user clicks on Edit Registration Information link from Patient dashboard")
+	@And("a user clicks on Edit Registration Information link from Patient dashboard")
 	public void loadRegistrationSummaryPage() {
-		registrationSummaryPage = (RegistrationSummaryPage) dashboardPage.goToRegistrationSummary();
+		registrationEditSectionPage = dashboardPage.clickOnEditPatient();
 	}
 
 	@Then("the system loads Registration Summary Page")
 	public void systemLoadsRegistrationSummaryPage() {
-		assertPage(registrationSummaryPage.waitForPage());
-	}
-
-	@When("a user clicks on Edit link from Registration Summary Page")
-	public void loadEditDemographicsSection() {
-		registrationEditSectionPage = (RegistrationEditSectionPage) registrationSummaryPage.clickOnEditDemographics();
+		assertTrue(textExists("Demographics"));
 	}
 
-	@Then("the system loads edit demographics section")
-	public void systemLoadsEditDemographicsSection() {
-		assertPage(registrationEditSectionPage.waitForPage());
-	}
-
-	@And("a user edits demographics")
+	@When("a user edits demographics")
 	public void editDemographics() {
-		testPatient.givenName = "Yuan";
-		testPatient.middleName = "Youn";
-		testPatient.familyName = "Ching";
 		testPatient.gender = "Male";
-		testPatient.birthDay = "23";
+		testPatient.birthDay = "21";
```

Comment:
any reason why changing the value of the birth day?

Reply:
Gonna erectify that